discovery/azure/azure.go

```
@@ -157,29 +159,45 @@ type azureClient struct {
 	vmssvm compute.VirtualMachineScaleSetVMsClient
 }
 
+// getAzureEnvironment returns an instance of azure.Environment based on envName.
+// If an invalid/unrecognized value of envName is passed, azurePublicCloud is returned by default.
```

This should be an error, not silently failing.